src/analyser.cpp

```diff
@@ -22,68 +22,72 @@ limitations under the License.
 #include "libcellml/analysermodel.h"
 #include "libcellml/analyservariable.h"
 #include "libcellml/component.h"
+#include "libcellml/generator.h"
+#include "libcellml/generatorprofile.h"
 #include "libcellml/model.h"
 #include "libcellml/units.h"
 #include "libcellml/validator.h"
 #include "libcellml/variable.h"
 
 #include <cmath>
+#include <iterator>
 
 #include "analyserequation_p.h"
 #include "analyserequationast_p.h"
 #include "analysermodel_p.h"
 #include "analyservariable_p.h"
 #include "anycellmlelement_p.h"
 #include "commonutils.h"
-#include "generator_p.h"
 #include "issue_p.h"
 #include "logger_p.h"
 #include "utilities.h"
 #include "xmldoc.h"
 #include "xmlutils.h"
 
-#ifdef TRUE
-#    undef TRUE
-#endif
-
-#ifdef FALSE
-#    undef FALSE
-#endif
-
-#ifdef NAN
-#    undef NAN
-#endif
+#include "libcellml/undefines.h"
 
 namespace libcellml {
 
 struct AnalyserInternalEquation;
 struct AnalyserInternalVariable;
 
+using AnalyserInternalEquationWeakPtr = std::weak_ptr<AnalyserInternalEquation>;
 using AnalyserInternalEquationPtr = std::shared_ptr<AnalyserInternalEquation>;
 using AnalyserInternalVariablePtr = std::shared_ptr<AnalyserInternalVariable>;
 
+using AnalyserInternalEquationWeakPtrs = std::vector<AnalyserInternalEquationWeakPtr>;
+using AnalyserInternalEquationPtrs = std::vector<AnalyserInternalEquationPtr>;
+using AnalyserInternalVariablePtrs = std::vector<AnalyserInternalVariablePtr>;
+
+using AnalyserEquationPtrs = std::vector<AnalyserEquationPtr>;
+using AnalyserVariablePtrs = std::vector<AnalyserVariablePtr>;
+using AnalyserExternalVariablePtrs = std::vector<AnalyserExternalVariablePtr>;
+
 struct AnalyserInternalVariable
 {
     enum struct Type
     {
-        UNKNOWN,
-        SHOULD_BE_STATE,
-        VARIABLE_OF_INTEGRATION,
-        STATE,
-        CONSTANT,
-        COMPUTED_TRUE_CONSTANT,
-        COMPUTED_VARIABLE_BASED_CONSTANT,
-        ALGEBRAIC,
-        OVERCONSTRAINED
+        UNKNOWN, // 0
+        SHOULD_BE_STATE, // 1
+        INITIALISED, // 2
+        VARIABLE_OF_INTEGRATION, // 3
+        STATE, // 4
+        CONSTANT, // 5
+        COMPUTED_TRUE_CONSTANT, // 6
+        COMPUTED_VARIABLE_BASED_CONSTANT, // 7
+        INITIALISED_ALGEBRAIC, // 8
+        STATE_ALGEBRAIC, // 9
+        ALGEBRAIC, // 10
+        OVERCONSTRAINED // 11
```

Comment:
Why add the integer values?  I can see no reason for it. Was it for debugging purposes?

Reply:
Yes, this is an internal type, so whenever I debugged things, I would print its `int` value and after a while it drove me nuts enough that I decided to put their `int` values.

Reply:
I do think that these comments should now be removed. I would suggest a debug utility if debugging help was still needed.

Reply:
Fair enough, will add a debug utility the next time I am working on this and need to do some debugging.

In the meantime: 36cd3691.